Code/Mantid/Framework/Geometry/inc/MantidGeometry/Instrument/Component.h

```diff
@@ -222,7 +222,20 @@ class MANTID_GEOMETRY_DLL Component : public virtual IComponent {
     else
       return std::string("");
   }
-
+  /** Get this component parameter's description -- no recursive search within children*/
```

Comment:
The names of these methods don't properly reflect what they do. For example, `Component::getDescription` would imply that the description is about the component, not the parameter. 

Also, why does `getDescription` not take a string `name` parameter and why do the recursive versions have `Par`  in the name and not the others?

From my perspective, would these not be a better set of methods:
  * `getParameterDescription(const std::string & name, bool recursive = true) const;`
  * `getShortParameterDescription(const std::string & name, bool recursive = true) const;`
  * `setParameterDescription(const std::string & name, const std::string & descr);`